### DISCUSSION TITLE
No Spaces in Lyrics

### DISCUSSION THREAD

**Discussion by JeffRocchio** [pedrolcl/dmidiplayer#34] [General]
I have a MIDI file with lyrics as text messages. They are displayed and sync properly in the playback; but there are no blank spaces between each word/syllable. For my use case this isn't a critical issue, but it would be nice to have the spaces there.

Screen Shot:
![Lyrics-Screenshot_20241228](https://github.com/user-attachments/assets/b2e38d5d-aa86-445d-91aa-5569161d9e4d)

Source MIDI File @ https://www.dropbox.com/scl/fi/iim4r4ajx4vv64rb1zpzt/Test-10_TblEdit_Charleston-Gals.mid?rlkey=22t2umj6vcgxlckgpye9s05lc&st=waqwn6ty&dl=0

**Comment by pedrolcl** [comment]
I cannot download your file. Please don't provide links to dropbox, or any other service that demands credentials. Instead, you should zip your samples and attach the zip file directly to the issue.

Anyway: spaces between words need to be provided by the file itself. If the SMF does not include separations between words or verses, nothing can be done by the player program. Please play and compare your file with some other player like VanBasco, or any other player, and provide a lyrics screenshot.

**Comment by pedrolcl** [comment]
It doesn't matter. I've got the file from your GitHub account here: https://github.com/JeffRocchio/MIDI-Count-In/blob/main/TestScores-and-MIDIs/Test-10_TblEdit_Charleston-Gals.mid . I assume that it is the same file.

This is what VanBasco can interpret from your song:

![image](https://github.com/user-attachments/assets/ba4bffa1-b3cd-45e4-8d26-f3b7786bd987)

As you can see, VanBasco adds a space between all syllables. Maybe this is what you wish?  When a song has the correct spaces attached to the end of each word this is not needed at all. Obviously, the player cannot know where the words (or the verses) end, but I suppose that this is better than displaying all the syllables together. 

Another player that you can try is falcosoft:

![image](https://github.com/user-attachments/assets/06822cf9-10f1-4bc3-9fa9-2cdf2e78f17e)

It has similar results. My suggestion is that you fix the song, adding a space to the end of each word and also a line break at the end of the verse.

**Comment by JeffRocchio** [comment]
I apologize about the Dropbox link. It was supposed to be a link that did
not require credentials to obtain the file, but apparently I hit the wrong
button. In any case you did get the correct file from my GitHub.

>*VanBasco adds a space between all syllables. Maybe this is what you wish?*<--
Yes, correct.

I've tried several other players on my Android phone and my iPad tablet.
All but one does add spaces when showing that file.

The file in question is generated out of the sheet music app TablEdit. So I
can't control its output. My python script is merely taking its output midi
export and adding my count-in bits to it's beginning. Tho in principle I
could walk through all the text messages and add a space to them as I am
adding my count part.

I can also open a request with the author of the TablEdit app and see if he
will make a modification so when it exports the lyrics it puts the spaces
in.

**Reply by pedrolcl** [reply]
> *VanBasco adds a space between all syllables. Maybe this is what you wish?*<--
> Yes, correct. 

Well, that is not the goal of Drumstick MIDI Player. Sorry. I want it to display the lyrics as they would be printed on any lyrics book: with spaces only between the words, and each verse on its own line, and lines grouped in stanzas. In other words: following the traditional formatting of poetry. But for this, the authors of the MIDI files need to follow the same conventions.

> I've tried several other players on my Android phone and my iPad tablet. All but one does add spaces when showing that file.

Then, you should use the player that outputs the lyrics text as you want.

> The file in question is generated out of the sheet music app TablEdit. So I can't control its output. My python script is merely taking its output midi export and adding my count-in bits to it's beginning. Tho in principle I could walk through all the text messages and add a space to them as I am adding my count part. I can also open a request with the author of the TablEdit app and see if he will make a modification so when it exports the lyrics it puts the spaces

I've never used TablEdit, but I don't understand the problem. Why you can't append the spaces yourself to the syllables where you want? What is special about the space? To me is just another character, like any letter or punctuation sign.

**Reply by JeffRocchio** [reply]
> Well, that is not the goal of Drumstick MIDI Player. 

Ok, I totally respect that Pedro; it makes perfect sense.

Re Table edit: The TablEdit files I'm getting are from other people so I truly don't have any control over them. 

But as I noted this is a relatively minor issue for me. I'm not trying to use these midi files for karaoke, just to learn and practice playing songs. Sometimes having lyrics is helpful, but not essential.

Thanks for listening and responding!
